### ISSUE TITLE
why sha256 not the HashFun?

### ISSUE THREAD

**Issue by yandrey** [helium/merkerl#7]
https://github.com/helium/merkerl/blob/26ddcaf7f3c2c76eebf6f9258822f923ce69cb75/src/merkerl.erl#L78

**Comment by madninja** [comment]
To avoid assumptions about HashFun in making intermediate nodes

**Comment by yandrey** [comment]
Sorry I don't understand your answer. That is, how do you mean "pluggable hash functions"? For example this has pluggable hash function https://github.com/merkletreejs/merkletreejs, but merkerl has fixed sha256

**Comment by madninja** [comment]
The sha256 is just used to identify the node and is the sha over the results of using the HashFun. The HashFun is actually used for the leaves
